draco/analysis/delay.py

```diff
@@ -846,34 +889,126 @@ def fourier_matrix_c2r(N, fsel=None):
     return Fr
 
 
+def fourier_matrix_c2c(N, fsel=None):
+    """Generate a Fourier matrix to represent a complex to complex FFT.
+
+    These Fourier conventions match `numpy.fft.fft()`.
+
+    Parameters
+    ----------
+    N : integer
+        Length of timestream that we are transforming to.
+    fsel : array_like, optional
+        Indices of the frequency channels to include in the transformation
+        matrix. By default, assume all channels.
+
+    Returns
+    -------
+    F : np.ndarray
+        An array performing the Fourier transform from a complex time series to
+        frequencies, with both input and output packed as alternating real and
+        imaginary elements.
+    """
+
+    if fsel is None:
+        fa = np.arange(N)
+    else:
+        fa = np.array(fsel)
+
+    fa = fa[:, np.newaxis]
+    ta = np.arange(N)[np.newaxis, :]
+
+    F = np.zeros((2 * fa.shape[0], 2 * N), dtype=np.float64)
+
+    arg = 2 * np.pi * ta * fa / N
+    F[0::2, 0::2] = np.cos(arg)
+    F[0::2, 1::2] = np.sin(arg)
+    F[1::2, 0::2] = -np.sin(arg)
+    F[1::2, 1::2] = np.cos(arg)
+
+    return F
+
+
+def _complex_to_alternating_real(array):
+    """View complex numbers as an array with alternating real and imaginary components.
+
+    Parameters
+    ----------
+    array : array_like
+        Input array of complex numbers.
+
+    Returns
+    -------
+    out : array_like
+        Output array of alternating real and imaginary components. These components are
+        expanded along the last axis, such that if `array` has `N` complex elements in
+        its last axis, `out` will have `2N` real elements.
+    """
+
+    return array.astype(np.complex128, order="C").view(np.float64).copy()
```

Comment:
Is it necessary to copy this here, rather than just returning the view. 

If you want to return a copy, I'd change the first line of the docstring to avoid the word view.